server/src/main/java/org/opensearch/gateway/remote/ClusterMetadataManifest.java

```diff
@@ -45,6 +45,7 @@ public class ClusterMetadataManifest implements Writeable, ToXContentFragment {
     // also we introduce index routing-metadata, diff and other attributes as part of manifest
     // required for state publication
     public static final int CODEC_V3 = 3; // In Codec V3, we have introduced new diff field in diff-manifest's routing_table_diff
+    public static final int CODEC_V4 = 4; // In Codec V4, we are adding checksum of cluster state.
```

Comment:
Do have integ tests which validate the codec version is handled properly? Also are we adding any integ test for this?

Reply:
Enabled existing ITs to run with checksum enabled. There are no ITs for codec version testing as of now. UTs are added for the new version.